Sources/AblyChat/RoomLifecycleManager.swift

```diff
@@ -1,8 +1,14 @@
-import Ably
+@preconcurrency import Ably
```

Comment:
do we need `@preconcurrency`?

Reply:
Pretty sure you get a compiler warning if you write `@preconcurrency` but there are no concurrency warnings to suppress, so I’d say yes

Reply:
Seems to be "Type `ARTChannelStateChange` does not conform to the `Sendable` protocol"... 

I thought we were aiming to remove all `@preconcurrency` tags?

Reply:
I think we can't avoid introducing them (temporarily) until the underlying issue gets investigated and fixed in ably-cocoa. I’ve changed this branch to be based on top of #81 (which switches to use `main`) and created https://github.com/ably/ably-cocoa/issues/1987 for addressing the remaining underlying issues causing the `@preconcurrency import Ably` introduced by this PR.